### ISSUE TITLE
docker slapd container permission issue

### ISSUE THREAD

**Issue by mkauzlar** [dcm4che/dcm4chee-arc-light#64]
Running the docker run --name slapd -p 389:389 .... as per instructions,
fails to start the slapd container with the following error:

56e94f86 @(#) $OpenLDAP: slapd  (Jan 16 2016 23:00:08) $
	root@chimera:/tmp/buildd/openldap-2.4.40+dfsg/debian/build/servers/slapd
56e94f86 ldif_read_file: Permission denied for "/etc/ldap/slapd.d/cn=config.ldif"
56e94f86 slapd stopped.
56e94f86 connections_destroy: nothing to destroy

**Comment by gunterze** [comment]
Not related to dcm4chee-arc-light - see https://github.com/docker/docker/issues/8460

**Comment by mkauzlar** [comment]
I'm sorry but I don't think is related. I'm running docker as root and your other containers start without issues.
Anyway it's not related to if docker is run as root but rather that the slapd container has permission issues inside the container itself (which should be checked upon container build).

**Comment by gunterze** [comment]
Right. Guess you run a newer image against a container external sldap data and configuration directory initiated by on older image. The user and group id of the slapd process changed. You have to chown the external directories to the new user id and group id, s. Dockerfile:

`# explicitly set user/group IDs RUN groupadd -r openldap --gid=1021 && useradd -r -g openldap --uid=1021 openldap`

**Comment by mkauzlar** [comment]
I checked and openldap user and group are already in the container.
The container is missing the file /etc/ldap/slapd.d/cn=config.ldif  which means openldap is not configured 
I deleted all images and containers and started from scratch. The error persists as ldap won't start in the container. I don't think ldap is installed in the image:
root@7c15648200e8:/# ls -l /usr/sbin/slapd
ls: cannot access /usr/sbin/slapd: No such file or directory

**Comment by gunterze** [comment]
I cannot reproduce the error with the latest image, neither of  dcm4che/slapd nor with dcm4che/slapd-dcm4chee:
```
> docker run --name slapd  -p 389:389 -d dcm4che/slapd-dcm4chee
d5b74807252acf1e20d71b5db7af9a722971577ce77e4332738ba0a5dc9e5d15
> docker ps
CONTAINER ID        IMAGE                    COMMAND                  CREATED             STATUS              PORTS                  NAMES
d5b74807252a        dcm4che/slapd-dcm4chee   "/docker-entrypoint.s"   24 seconds ago      Up 23 seconds       0.0.0.0:389->389/tcp   slapd
> docker logs slapd
....
adding new entry "cn=Verification SOP Class SCU,dicomAETitle=DCM4CHEE_TRASH,dicomDeviceName=dcm4chee-arc,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org"

SASL/EXTERNAL authentication started
SASL username: gidNumber=0+uidNumber=0,cn=peercred,cn=external,cn=auth
SASL SSF: 0
modifying entry "dicomDeviceName=dcm4chee-arc,cn=Devices,cn=DICOM Configuration,dc=dcm4che,dc=org"

56eaef31 @(#) $OpenLDAP: slapd  (Jan 16 2016 23:00:08) $
	root@chimera:/tmp/buildd/openldap-2.4.40+dfsg/debian/build/servers/slapd
56eaef31 slapd starting

```

**Comment by mkauzlar** [comment]
Yes your command: `docker run --name slapd  -p 389:389 -d dcm4che/slapd-dcm4chee` 
produces a running ldap, however the command from the instructions:

```
docker run --name slapd \
           -p 389:389 \
           -e LDAP_BASE_DN=dc=dcm4che,dc=org \
           -e LDAP_ORGANISATION=dcm4che.org \
           -e LDAP_ROOTPASS=secret \
           -e LDAP_CONFIGPASS=secret \
           -e DEVICE_NAME=dcm4chee-arc \
           -e AE_TITLE=DCM4CHEE \
           -e DICOM_HOST=dockerhost \
           -e DICOM_PORT=11112 \
           -e HL7_PORT=2575 \
           -e SYSLOG_HOST=logstash \
           -e SYSLOG_PORT=8512 \
           -e SYSLOG_PROTOCOL=UDP \
           -e STORAGE_DIR=/storage/fs1 \
           -v /var/local/dcm4chee-arc/ldap:/var/lib/ldap \
           -v /var/local/dcm4chee-arc/slapd.d:/etc/ldap/slapd.d \
           -d dcm4che/slapd-dcm4chee:5.1.0
```

doesn't work.
The problem is the line:` -v /var/local/dcm4chee-arc/slapd.d:/etc/ldap/slapd.d \`
I'm not sure of the implications of removing this line

**Comment by gunterze** [comment]
I also cannot reproduce the error when mapping the data and configuration directory to host directories. Are you sure you have not such directories from a previous installation owned by a different user? If so,  just delete them:
```
> rm -r /var/local/dcm4chee-arc/ldap
> rm -r /var/local/dcm4chee-arc/slapd.d
```
They will be recreated on container start owned with the new openldap user:
```
> ls -l /var/local/dcm4chee-arctotal 8
drwxr-sr-x 2 1021 1021 4096 Mär 17 22:23 ldap
drwxr-sr-x 3 1021 1021 4096 Mär 17 22:23 slapd.d
```

**Comment by mkauzlar** [comment]
As I said, I start from scratch (remove deinstall images and docker,
nothing is present) and then just install docker and the slapd container.
I tried this now on 2 PCs with the same result. Then I tried on my laptop
and it worked, no errors (??).
All systems are Ubuntu 14.04. The only difference is the first two have the
kernel 3.13.0-37-generic while the laptop has kernel 4.3.0-040300-generic
but I'm not sure how this matters.
The Docker version is the same on all systems.



On Thu, Mar 17, 2016 at 10:30 PM, Gunter Zeilinger <notifications@github.com
